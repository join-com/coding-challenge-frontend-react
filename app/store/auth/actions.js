import {
  SIGN_IN_REQUEST,
  SIGN_IN_SUCCESS,
  SIGN_OUT,
  SIGN_UP_REQUEST,
} from './constants';

export function submitSignInRequest(data, intl, messages) {
  return {
    type: SIGN_IN_REQUEST,
    data,
    intl,
    messages,
  };
}

export function setSignInData(user, token) {
  return {
    type: SIGN_IN_SUCCESS,
    user,
    token,
  };
}

export function signOut() {
  return {
    type: SIGN_OUT,
  };
}

export function submitSignUpRequest(data, intl, messages) {
  return {
    type: SIGN_UP_REQUEST,
    data,
    intl,
    messages,
  };
}
