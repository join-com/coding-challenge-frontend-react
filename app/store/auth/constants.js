export const SIGN_IN_FORM = 'forms/auth/SIGN_IN';
export const SIGN_UP_FORM = 'forms/auth/SIGN_UP';

export const SIGN_IN_REQUEST = 'actions/auth/SIGN_IN_REQUEST';
export const SIGN_IN_SUCCESS = 'actions/auth/SIGN_IN_SUCCESS';
export const SIGN_OUT = 'actions/auth/SIGN_OUT';
export const SIGN_UP_REQUEST = 'actions/auth/SIGN_UP_REQUEST';

export const ERRORS_PREFIX = 'boilerplate.shared.Auth.errors';

export const DUPLICATED_EMAIL = 'DUPLICATED_EMAIL';
export const EMPTY_PASSWORD = 'EMPTY_PASSWORD';
export const INCORRECT_CREDENTIALS = 'INCORRECT_CREDENTIALS';
export const INVALID_EMAIL = 'INVALID_EMAIL';
export const INVALID_NAME = 'INVALID_NAME';
export const INVALID_PASSWORD = 'INVALID_PASSWORD';
export const FORM_SUBMISSION_FAILED = 'FORM_SUBMISSION_FAILED';
