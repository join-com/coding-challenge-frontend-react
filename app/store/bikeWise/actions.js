import {
  LOAD_ITEMS,
  LOAD_ITEMS_ERROR,
  LOAD_ITEMS_SUCCESS,
} from './constants';

export function loadItems() {
  return {
    type: LOAD_ITEMS,
  };
}

export function setItems(items) {
  return {
    type: LOAD_ITEMS_SUCCESS,
    items,
  };
}

export function setLoadItemsError(error) {
  return {
    type: LOAD_ITEMS_ERROR,
    error,
  };
}
