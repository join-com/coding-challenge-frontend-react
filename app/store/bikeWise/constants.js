export const CRITERIA_FORM = 'forms/bikeWise/CRITERIA_FORM';

export const LOAD_ITEMS = 'actions/bikeWise/LOAD_ITEMS';
export const LOAD_ITEMS_ERROR = 'actions/bikeWise/LOAD_ITEMS_ERROR';
export const LOAD_ITEMS_SUCCESS = 'actions/bikeWise/LOAD_ITEMS_SUCCESS';
