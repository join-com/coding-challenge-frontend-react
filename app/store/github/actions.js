import {
  CHANGE_USERNAME,
  LOAD_REPOS,
  LOAD_REPOS_ERROR,
  LOAD_REPOS_SUCCESS,
} from './constants';

export function changeUsername(username) {
  return {
    type: CHANGE_USERNAME,
    username,
  };
}

export function loadRepositories() {
  return {
    type: LOAD_REPOS,
  };
}

export function setRepositories(repositories, username) {
  return {
    type: LOAD_REPOS_SUCCESS,
    repositories,
    username,
  };
}

export function setLoadRepositoriesError(error) {
  return {
    type: LOAD_REPOS_ERROR,
    error,
  };
}
