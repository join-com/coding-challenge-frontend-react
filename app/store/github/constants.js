export const REPOSITORIES_FORM = 'forms/github/REPOSITORIES_FORM';

export const CHANGE_USERNAME = 'actions/github/CHANGE_USERNAME';
export const LOAD_REPOS = 'actions/github/LOAD_REPOS';
export const LOAD_REPOS_ERROR = 'actions/github/LOAD_REPOS_ERROR';
export const LOAD_REPOS_SUCCESS = 'actions/github/LOAD_REPOS_SUCCESS';
