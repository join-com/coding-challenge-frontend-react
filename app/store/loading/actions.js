import {
  SET_LOADING,
  RESET_LOADING,
} from './constants';

export function resetLoading() {
  return {
    type: RESET_LOADING,
  };
}

export function setLoading() {
  return {
    type: SET_LOADING,
  };
}
