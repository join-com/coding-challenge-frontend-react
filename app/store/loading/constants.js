export const SET_LOADING = 'actions/loading/SET_LOADING';
export const RESET_LOADING = 'actions/loading/RESET_LOADING';
