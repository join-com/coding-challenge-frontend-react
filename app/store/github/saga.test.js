import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_REPOS } from './constants';
import { setRepositories, setLoadRepositoriesError } from './actions';
import { resetLoading } from '../loading/actions';

import generateRootSaga, { getRepositories } from './saga';

const username = 'alexander-elgin';

/* eslint-disable redux-saga/yield-effects */
describe('github Saga', () => {
  let getRepositoriesGenerator;

  beforeEach(() => {
    getRepositoriesGenerator = getRepositories();

    const selectDescriptor = getRepositoriesGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = getRepositoriesGenerator.next(username).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  describe('the repositories data are fetched successfully', () => {
    it('dispatches the setRepositories action', () => {
      const repositories = [
        {
          name: 'First repo',
        },
        {
          name: 'Second repo',
        },
      ];

      const putDescriptor = getRepositoriesGenerator.next(repositories).value;
      expect(putDescriptor).toEqual(put(setRepositories(repositories, username)));
    });
  });

  describe('the request fails', () => {
    it('dispatches the setLoadRepositoriesError action', () => {
      const response = new Error('Some error');
      const putDescriptor = getRepositoriesGenerator.throw(response).value;
      expect(putDescriptor).toEqual(put(setLoadRepositoriesError(response)));
    });
  });

  afterEach(() => {
    const putDescriptor = getRepositoriesGenerator.next().value;
    expect(putDescriptor).toEqual(put(resetLoading()));
  });
});

describe('root saga', () => {
  const rootSaga = generateRootSaga();

  it('should start task to watch for LOAD_REPOS action', () => {
    const takeLatestDescriptor = rootSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_REPOS, getRepositories));
  });
});
