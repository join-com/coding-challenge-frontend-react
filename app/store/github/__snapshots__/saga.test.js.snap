// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`github Saga the repositories data are fetched successfully dispatches the setRepositories action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`github Saga the repositories data are fetched successfully dispatches the setRepositories action 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/users/alexander-elgin/repos",
      Object {
        "sort": "updated",
        "type": "all",
      },
      "https://api.github.com",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`github Saga the request fails dispatches the setLoadRepositoriesError action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`github Saga the request fails dispatches the setLoadRepositoriesError action 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/users/alexander-elgin/repos",
      Object {
        "sort": "updated",
        "type": "all",
      },
      "https://api.github.com",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
